tests/plugins/widget/integration/wysiwygarea/superflousdoesntremovedivwrapper.js

```diff
@@ -0,0 +1,57 @@
+/* bender-ckeditor-plugins: widget,wysiwygarea,undo,toolbar */
+
+
+( function() {
+	'use strict';
+
+	bender.editor = {
+		config: {
+			allowedContent: true
+		}
+	};
+
+	bender.test( {
+		'test keeping widget wrapper in editor when superflous elemnts are checked': function() {
+			CKEDITOR.plugins.add( 'test', {
+				requires: 'widget',
+				init: function( editor ) {
+					editor.widgets.add( 'test', {
+						upcast: function( element ) {
+							if ( element.hasClass( 'test' ) ) {
+								return true;
+							}
+						},
+
+						init: function() {
+							this.element.setStyle( 'background-color', '#FFBB00' );
+							this.element.setStyle( 'padding', '30px' );
+						}
+					} );
+				}
+			} );
+			bender.editorBot.create( {
+				name: 'test_widget',
+				startupData: '<h2 class="test">Test</h2><p>Preheat the oven to</p>',
+				config: {
+					extraPlugins: 'test',
+					allowedContent: true
+				}
+			}, function( bot ) {
+				var editor = bot.editor;
+
+				editor.editable().fire( 'keydown', new CKEDITOR.dom.event( {
+					keyCode: 90,
+					ctrlKey: true,
+					shiftKey: false
+				} ) );
+
+				var snap = editor.getSnapshot();
+				editor.loadSnapshot( snap );
```

Comment:
Yes.
`keydown` and `keyup` don't trigger undo command, which triggers re-creating widget (there even is no history to revert it). 
So this simulation works as follows:
1. Simulate `keydown` to register listeners on `div` by superflous function.
2. Reload snapshot, to simulate <kbd>Ctrl</kbd>+<kbd>Z</kbd>, when previous snapshot is restored.
3. Reloading snapshot automatically triggers re-creating widget. Here will be destroyed wrapping div for widget, and `retain` attribute will be lost (as it happens in Edge).
4. Simulate `keyup` to execute proper listener from superflous function.

Re-loading snapshot might be replaced with some kind of triggering undo event, but it would require much more work. First to make some history, and then revert it. So I extract only important part from entire undo process, which is reloading snapshot, what cause the issue in case of using shortcut. This allows on proper simulation behaviour on Edge and proper fail of test in case of Edge environment without bug fix.